experimental/packages/otlp-exporter-base/src/retryable-transport.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IExporterTransport } from './exporter-transport';
+import { ExportResponse } from './export-response';
+
+const MAX_ATTEMPTS = 5;
+const INITIAL_BACKOFF = 1000;
+const MAX_BACKOFF = 5000;
+const BACKOFF_MULTIPLIER = 1.5;
+
+class RetryingTransport implements IExporterTransport {
+  constructor(private _transport: IExporterTransport) {}
+
+  private retry(data: Uint8Array, inMillis: number): Promise<ExportResponse> {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        this._transport.send(data).then(resolve, reject);
+      }, inMillis);
+    });
+  }
+
+  async send(data: Uint8Array): Promise<ExportResponse> {
+    let result = await this._transport.send(data);
+    let attempts = MAX_ATTEMPTS;
+    let nextBackoff = INITIAL_BACKOFF;
+
+    // TODO: I'm not 100% sure this is correct, please review in-depth.
+    while (result.status === 'retryable' && attempts > 0) {
+      attempts--;
+      const upperBound = Math.min(nextBackoff, MAX_BACKOFF);
+      const backoff = Math.random() * upperBound;
```

Comment:
IIUC this is an iterative way of implementing exponential backoff (with 1,5 as exponent). What I do not fully get is this multiplication with `Math.rand()` 🤔 

With this randomness in place the waiting time between retries may not increase. We may even get small values consuming the reties very fast.

- Maybe we need also a `lowerBound`?
- Or maybe its even simpler if just use `Math.min(nextBackoff, MAX_BACKOFF)`? The progression is [1000, 1500, 2250, 3375, 5000, 5000 ...]